### ISSUE TITLE
Scheduled events are not properly applied

### ISSUE THREAD

**Issue by lhilbert** [lhilbert/active-microemulsion#36]
1) Application of events (activate, flavopiridol, ten-spike etc) that are passed as flags and also recognized is not consistent. They sometimes simply do not get applied, in a not predictable manner. This seems to get better when we switch from minute units to second units, so it might be an inconsistent application of the scaling factor.

2) The parameter change events are always delayed until the next time an image is taken. So it seems they are stacked into the same chain of events as the image taking events, but the image taking events stall the execution of the parameter change events. Maybe the order of events needs to be sorted at some point in the code everytime before checking if an event is to  be executed?

**Comment by churli** [comment]
Can you please provide an example of a command line that displays the issue? I would like to try to reproduce it so I can debug it more easily ;)
Thanks!

**Comment by lhilbert** [comment]
Yes, sure. For the point (2), the behavior needs no special settings, any run where a parameter change event is scheduled at a different time than a snapshot image will show the phenomenon that the change waits for the snapshot.

For point (1), I will describe the behavior when the -m flag is passed. When the -m flag is not passed, it does not occur. When I put only a single event ```--flavopiridol 0```, this is applied. When I now put an additional event, or also a flavopiridol event with greater than 0 time, not all the events are executed. A good way to verify this is by running a very small test simulation, with very low swapping rate.

Does this help?
